### ISSUE TITLE
Error when adding to config (no attribute TMCtstepHelper)

### ISSUE THREAD

**Issue by Physicsly** [andrewmcgr/klipper_tmc_autotune#207]
When I add the line #[autotune_tmc stepper_x] to my printer.cfg file, I get this error 

Internal error during connect: module 'extras.tmc' has no attribute 'TMCtstepHelper'

My printer is a Creality CR-10 SE

printer.cfg:
https://pastebin.com/pU8nMJFg

from klippy.log (it was too big to put in a pastebin)
[INFO] 2024-10-17 00:23:53,979 [root] [gcode_move:__init__:26] self.variable_safe_z = 3.0
[ERROR] 2024-10-17 00:23:56,827 [root] [klippy:_connect:248] Unhandled exception during connect
Traceback (most recent call last):
  File "/usr/share/klipper/klippy/klippy.py", line 193, in _connect
    self._read_config()
  File "/usr/share/klipper/klippy/klippy.py", line 159, in _read_config
    self.load_object(config, section_config.get_name(), None)
  File "/usr/share/klipper/klippy/klippy.py", line 126, in load_object
    self.objects[section] = init_func(config.getsection(section))
  File "/usr/share/klipper/klippy/extras/autotune_tmc.py", line 393, in load_config_prefix
    return AutotuneTMC(config)
  File "/usr/share/klipper/klippy/extras/autotune_tmc.py", line 99, in __init__
    if 'pstepper' in signature(tmc.TMCtstepHelper).parameters:
AttributeError: module 'extras.tmc' has no attribute 'TMCtstepHelper'

**Comment by jjiill888** [comment]
Hi my printer is k1se, and I have same problem.

The reason that I can't find 'TMCtstepHelper' in my printer's klippy/extras/tmc.py, but I can find that in newest version [klipper/extras/tmc.py](https://github.com/search?q=repo%3AKlipper3d%2Fklipper%20TMCtstepHelper&type=code). 

So I think creatily klipper's version are too old to support klipper_tmc_autotune and am going to fix that

**Comment by Physicsly** [comment]
Yep, I tried updating the python file but it bricked my printer and disabled ssh. I had to use the k1 exploit to send a factory reset packet to it.

**Comment by jjiill888** [comment]
It's listening to a bad experience. I hope adding new configurations to tmc.py can help you solve these problems. But I found a new issue: Creality printers always use lower torque motors (400 Nm) for the A/B/Z axes, and step loss occurs in the default configuration when tmc_autotune is applied, and It seems that the default configuration of the project may have a negative effect on motors with low holding torque. I have already forked this project and am adapting it for the Creality K1SE. This may be helpful for other Creality machines using this script [This is my brench](https://github.com/jjiill888/klipper_tmc_autotune/tree/k1c/k1se)

**Comment by jjiill888** [comment]
I’ve figured out why the issue occurs. The reason is that Creality's Klipper has been heavily trimmed, and 'TMCtstepHelper' has been removed from tmc.py. You must place the TMCtstepHelper implementation right before the TMCStealthchopHelper in tmc.py. For more details, check out my branch. I sincerely hope this helps resolve your issue. [my_brench](https://github.com/jjiill888/klipper_tmc_autotune/tree/k1c/k1se)

**Comment by Physicsly** [comment]
I will test this once I get to it
